### ISSUE TITLE
curl return code is 0 if download file is redirected 

### ISSUE THREAD

**Issue by sophia-guo** [AdoptOpenJDK/openjdk-tests#582]
curl return code is 0 if download file is redirected. 
```
curl -OLJks ${curl_options} $file
		if [ $? -ne 0 ]; then
	...

	fi
```

The code need to update

12:24:42 curl -OLJks  https://api.adoptopenjdk.net/v2/binary/nightly/openjdk8?openjdk_impl=openj9&os=linux&arch=x64&release=latest&type=jdk 12:24:42 unzip file: openjdk8?openjdk_impl=openj9&os=linux&arch=x64&release=latest&type=jdk ... 12:24:42 12:24:42 gzip: openjdk8?openjdk_impl=openj9&os=linux&arch=x64&release=latest&type=jdk: not in gzip format 12:24:42 tar: This does not look like a tar archive 12:24:42 tar: Exiting with failure status due to previous errors 12:24:42 ls: cannot access '*/': No such file or directory 12:24:42 chmod: cannot access 'j2sdk-image': No such file or directory
--

**Comment by smlambert** [comment]
Looks like this is fixed: https://github.com/AdoptOpenJDK/openjdk-tests/blob/master/get.sh#L163
@sophia-guo can this be closed?

**Comment by sophia-guo** [comment]
No. The issue is with redirect option -O even if the URL is wrong(or no available jar) the command return code is still 0 -- meaning everything is ok. Normally without -O if error happens the return code is non zero -- which can be caught by `if [ $? -ne 0 ]` .

**Comment by smlambert** [comment]
So another check is required even in the "success" returns 0 case, to see that we get an SDK to test ?

**Comment by sophia-guo** [comment]
This is fixed by #1099 and close it.
